tests/congruence_tests/test_query.py

```diff
@@ -0,0 +1,1094 @@
+
+    for i in range(100):
+        query_filter = one_random_filter_please()
+        try:
+            compare_client_results(
+                local_client,
+                remote_client,
+                searcher.dense_queries_prefetch_filtered,
+                query_filter=query_filter,
+            )
+        except AssertionError as e:
+            print(f"\nAttempt {i} failed with filter {query_filter}")
+            raise e
+
+
+def test_dense_query_prefetch_score_threshold():
+    fixture_points = generate_fixtures()
+
+    searcher = TestSimpleSearcher()
+
+    local_client = init_local()
+    init_client(local_client, fixture_points)
+
+    remote_client = init_remote()
+    init_client(remote_client, fixture_points)
+
+    compare_client_results(
+        local_client, remote_client, searcher.dense_queries_prefetch_score_threshold
+    )
+
+
+def test_dense_query_prefetch_parametrized():
+    fixture_points = generate_fixtures()
+
+    searcher = TestSimpleSearcher()
+
+    local_client = init_local()
+    init_client(local_client, fixture_points)
+
+    remote_client = init_remote()
+    init_client(remote_client, fixture_points)
+
+    compare_client_results(
+        local_client,
+        remote_client,
+        searcher.dense_queries_prefetch_parametrized,
+        search_params={"exact": True},
+    )
+    compare_client_results(
+        local_client,
+        remote_client,
+        searcher.dense_queries_prefetch_parametrized,
+        search_params={"hnsw_ef": 128},
+    )
+    compare_client_results(
+        local_client,
+        remote_client,
+        searcher.dense_queries_prefetch_parametrized,
+        search_params={"indexed_only": True},
+    )
+    compare_client_results(
+        local_client,
+        remote_client,
+        searcher.dense_queries_prefetch_parametrized,
+        search_params={"quantization": {"ignore": True, "rescore": True, "oversampling": 2.0}},
+    )
+
+
+def test_dense_query_parametrized():
+    fixture_points = generate_fixtures()
+
+    searcher = TestSimpleSearcher()
+
+    local_client = init_local()
+    init_client(local_client, fixture_points)
+
+    remote_client = init_remote()
+    init_client(remote_client, fixture_points)
+
+    compare_client_results(
+        local_client,
+        remote_client,
+        searcher.dense_queries_parametrized,
+        search_params={"exact": True},
+    )
+    compare_client_results(
+        local_client,
+        remote_client,
+        searcher.dense_queries_parametrized,
+        search_params={"hnsw_ef": 128},
+    )
+    compare_client_results(
+        local_client,
+        remote_client,
+        searcher.dense_queries_parametrized,
+        search_params={"indexed_only": True},
+    )
+    compare_client_results(
+        local_client,
+        remote_client,
+        searcher.dense_queries_parametrized,
+        search_params={"quantization": {"ignore": True, "rescore": True, "oversampling": 2.0}},
+    )
```

Comment:
local mode does not work with search params
remote mode on the amount of points we have (1000) does not build an hnsw index

all the comparisons are actually equal, we can just leave one set of params in order to check the interface

Reply:
I am not sure how the test with "indexed_only": True actually passes, since remote client does not have any indexed points and should not return any points

Reply:
Apparently, it returns them

Reply:
The answer:
everything which is below indexing_threshold will still be returned even if `indexed_only: True`